Sources/ReleaseServer/ReleaseServer.WebApi/Repositories/ReleaseArtifactRepository.cs

```diff
@@ -200,6 +199,53 @@ public List<string> GetPlatforms(string productName, string version)
 
             return platforms.OrderBy(p => p).ToList();
         }
+        
+        public BackupInformationModel RunBackup()
+        {
+            var timeStamp = DateTime.UtcNow.ToString("yyyy-MM-dd-HH-mm-ss");
+            
+            string backupFileName = "backup_" + timeStamp + ".zip";
+            string backupArchiveFileName = Path.Combine(BackupRoot, backupFileName); 
+            
+            //Clear the backup folder first
+            DeleteFilesInPath(BackupRoot);
+            
+            //Create the backup -> zip the whole ArtifactRoot folder
+            ZipFile.CreateFromDirectory(ArtifactRoot, backupArchiveFileName);
+
+            return new BackupInformationModel
+            {
+                FullPath = backupArchiveFileName,
+                FileName = backupFileName
+            };
+        }
+
+        public void RestoreBackup(ZipArchive backupPayload)
+        {
+            try
+            {
+                //Clear the whole artifact root directory
+                Directory.Delete(ArtifactRoot, true);
```

Comment:
This will throw a `System.IO.DirectoryNotFoundException` if the `ArtifactRoot` directory does not exist.
Nevertheless this would result in an error when running inside a docker container where is directory has been bind mounted from the host.

I would suggest to check if the directory exists and delete the content if it exists or create it.
For deleting the content an extension method `DeleteContent()` of `DirectoryInfo` can be implemented (analogous to `DeleteFiles()` described on line 211).

Reply:
Realized, see: 

https://github.com/Traeger-GmbH/release-server/blob/feature/implement-backup-mechanism/Sources/ReleaseServer/ReleaseServer.WebApi/Repositories/ReleaseArtifactRepository.cs#L241-L251